parquet-hadoop tests fail if working behind a web proxy

### Describe the enhancement requested

My working environment is behind a web proxy. The maven build for me fails on `parquet-hadoop` tests that use files from `https://github.com` which is not directly accessible to me.

The failed tests are:
```
[ERROR]   TestInterOpReadByteStreamSplit.testReadAllSupportedTypes » SocketTimeout Connect timed out
[ERROR]   TestInterOpReadByteStreamSplit.testReadFloats » SocketTimeout Connect timed out
[ERROR]   TestInterOpReadFloat16.testInterOpReadFloat16NonZerosAndNansParquetFiles » SocketTimeout Connect timed out
[ERROR]   TestInterOpReadFloat16.testInterOpReadFloat16ZerosAndNansParquetFiles » SocketTimeout Connect timed out
[ERROR]   TestInteropBloomFilter.testReadDataIndexBloomParquetFiles » SocketTimeout Connect timed out
[ERROR]   TestInteropBloomFilter.testReadDataIndexBloomWithLengthParquetFiles » SocketTimeout Connect timed out
[ERROR]   TestInteropReadLz4RawCodec.testInteropReadLz4RawLargerParquetFiles » SocketTimeout Connect timed out
[ERROR]   TestInteropReadLz4RawCodec.testInteropReadLz4RawSimpleParquetFiles » SocketTimeout Connect timed out
```

They all are similar:
```
org.apache.parquet.hadoop.TestInterOpReadByteStreamSplit.testReadAllSupportedTypes -- Time elapsed: 10.24 s <<< ERROR!
java.net.SocketTimeoutException: Connect timed out
        at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:551)
        at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:602)
        at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
        at java.base/java.net.Socket.connect(Socket.java:633)
        at okhttp3.internal.platform.Platform.connectSocket(Platform.kt:128)
        at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:295)
        at okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207)
        at okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226)
```


The main reason for this is that `OkHttpClient` for these tests is not configured to use any proxy, and it does not recognize system properties (e.g. `https.proxyHost`) or environment variables (e.g. `https_proxy`) for such flow.


### Component(s)

Build

--- a/parquet-hadoop/src/test/java/org/apache/parquet/hadoop/InterOpTester.java
+++ b/parquet-hadoop/src/test/java/org/apache/parquet/hadoop/InterOpTester.java
@@ -38,8 +38,14 @@ public class InterOpTester {
   private static final Logger LOG = LoggerFactory.getLogger(InterOpTester.class);
   // since PARQUET_TESTING_REPO might be beyond a web proxy ...
   private static OkHttpClient createOkHttpClientOptProxy() {
-    String proxyHost = System.getProperty("https.proxyHost");
-    String proxyPort = System.getProperty("https.proxyPort");
+    /* We use a different JVM property set,
+     * because CI may define JVM properties
+     * "https.proxyHost" and "https.proxyPort"
+     * and that proxy won't support some compressions
+     * (e.g. gzip/snappy on github.com CI).
+     /
+    String proxyHost = System.getProperty("parquet.https.proxyHost");
+    String proxyPort = System.getProperty("parquet.https.proxyPort");
     OkHttpClient client = null;
     if (proxyHost != null || proxyPort != null) {
       try {

--- a/parquet-hadoop/src/test/java/org/apache/parquet/hadoop/InterOpTester.java
+++ b/parquet-hadoop/src/test/java/org/apache/parquet/hadoop/InterOpTester.java
@@ -37,7 +37,7 @@ public class InterOpTester {
   private static final String PARQUET_TESTING_PATH = "target/parquet-testing/";
   private static final Logger LOG = LoggerFactory.getLogger(InterOpTester.class);
   // since PARQUET_TESTING_REPO might be beyond a web proxy ...
-  private static OkHttpClient createOkHttpClientOptProxy() {
+  public static OkHttpClient createOkHttpClientOptProxy() {
     /* We use a different JVM property set,
      * because CI may define JVM properties
      * "https.proxyHost" and "https.proxyPort"

--- a/parquet-hadoop/src/test/java/org/apache/parquet/hadoop/InterOpTester.java
+++ b/parquet-hadoop/src/test/java/org/apache/parquet/hadoop/InterOpTester.java
@@ -47,7 +47,7 @@ public class InterOpTester {
     String proxyHost = System.getProperty("parquet.https.proxyHost");
     String proxyPort = System.getProperty("parquet.https.proxyPort");
     OkHttpClient client = null;
-    if (proxyHost != null || proxyPort != null) {
+    if (proxyHost != null && proxyPort != null) {
       try {
         int port = Integer.valueOf(proxyPort);
         Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyHost, port));

--- a/parquet-hadoop/src/test/java/org/apache/parquet/hadoop/InterOpTester.java
+++ b/parquet-hadoop/src/test/java/org/apache/parquet/hadoop/InterOpTester.java
@@ -43,7 +43,7 @@ public class InterOpTester {
      * "https.proxyHost" and "https.proxyPort"
      * and that proxy won't support some compressions
      * (e.g. gzip/snappy on github.com CI).
-     /
+     */
     String proxyHost = System.getProperty("parquet.https.proxyHost");
     String proxyPort = System.getProperty("parquet.https.proxyPort");
     OkHttpClient client = null;

--- a/parquet-hadoop/src/test/java/org/apache/parquet/hadoop/InterOpTester.java
+++ b/parquet-hadoop/src/test/java/org/apache/parquet/hadoop/InterOpTester.java
@@ -20,6 +20,8 @@
 package org.apache.parquet.hadoop;
 
 import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Proxy;
 import okhttp3.OkHttpClient;
 import okhttp3.Request;
 import okhttp3.Response;
@@ -34,7 +36,24 @@ public class InterOpTester {
   private static final String PARQUET_TESTING_REPO = "https://github.com/apache/parquet-testing/raw/";
   private static final String PARQUET_TESTING_PATH = "target/parquet-testing/";
   private static final Logger LOG = LoggerFactory.getLogger(InterOpTester.class);
-  private OkHttpClient httpClient = new OkHttpClient();
+  // since PARQUET_TESTING_REPO might be beyond a web proxy ...
+  private static OkHttpClient createOkHttpClientOptProxy() {
+    String proxyHost = System.getProperty("https.proxyHost");
+    String proxyPort = System.getProperty("https.proxyPort");
+    OkHttpClient client = null;
+    if (proxyHost != null || proxyPort != null) {
+      try {
+        int port = Integer.valueOf(proxyPort);
+        Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyHost, port));
+        client = new OkHttpClient.Builder().proxy(proxy).build();
+      } catch (NumberFormatException e) {
+      }
+    }
+    if (client == null) client = new OkHttpClient();
+    return client;
+  }
+
+  private OkHttpClient httpClient = createOkHttpClientOptProxy();
 
   public Path GetInterOpFile(String fileName, String changeset) throws IOException {
     return GetInterOpFile(fileName, changeset, "data");

--- a/parquet-hadoop/src/test/java/org/apache/parquet/hadoop/InterOpTester.java
+++ b/parquet-hadoop/src/test/java/org/apache/parquet/hadoop/InterOpTester.java
@@ -53,6 +53,13 @@ public class InterOpTester {
         Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyHost, port));
         client = new OkHttpClient.Builder().proxy(proxy).build();
       } catch (NumberFormatException e) {
+        LOG.warn(
+            "Ignoring proxy configuration because proxy port could not be parsed: "
+                + "parquet.https.proxyHost='{}', parquet.https.proxyPort='{}'. "
+                + "Falling back to a direct connection.",
+            proxyHost,
+            proxyPort,
+            e);
       }
     }
     if (client == null) client = new OkHttpClient();

--- a/parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestInteropBloomFilter.java
+++ b/parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestInteropBloomFilter.java
@@ -57,7 +57,7 @@ public class TestInteropBloomFilter {
   private static String DATA_INDEX_BLOOM_WITH_LENGTH_FILE = "data_index_bloom_encoding_with_length.parquet";
 
   private static final Logger LOG = LoggerFactory.getLogger(TestInteropBloomFilter.class);
-  private OkHttpClient httpClient = new OkHttpClient();
+  private OkHttpClient httpClient = InterOpTester.createOkHttpClientOptProxy();
 
   @Test
   public void testReadDataIndexBloomParquetFiles() throws IOException {

